src/main/java/org/folio/circulation/domain/policy/LostItemPolicyRepository.java

```diff
@@ -0,0 +1,104 @@
+package org.folio.circulation.domain.policy;
+
+import io.vertx.core.json.JsonObject;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.FetchSingleRecord;
+import org.folio.circulation.support.MultipleRecordFetcher;
+import org.folio.circulation.support.Result;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static org.folio.circulation.support.Result.ofAsync;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ResultBinding.mapResult;
+
+public class LostItemPolicyRepository extends CirculationPolicyRepository<LostItemPolicy> {
+
+  public LostItemPolicyRepository(Clients clients) {
+    super(clients.circulationLostItemRules(), clients.lostItemPoliciesStorage());
+  }
+
+  public CompletableFuture<Result<LoanAndRelatedRecords>> lookupLostItemPolicy(
+          LoanAndRelatedRecords relatedRecords) {
+
+    return Result.of(relatedRecords::getLoan)
+            .combineAfter(this::lookupPolicy, Loan::withLostItemPolicy)
+            .thenApply(mapResult(relatedRecords::withLoan));
```

Comment:
Why are these method invocations nested so far in?

Reply:
To set a new loan to LoanAndRelatedRecords entity

Reply:
What I'm wondering is why are these lines indented by more than 2 or 4 spaces?

Reply:
Fixed